core/src/main/java/io/grpc/internal/ExponentialBackoffPolicy.java

```diff
@@ -39,26 +39,52 @@
 import java.util.concurrent.TimeUnit;
 
 /**
- * Retry Policy for Transport reconnection.  Initial parameters from
- * https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
- *
- * <p>TODO(carl-mastrangelo): add unit tests for this class
+ * Exponential backoff policy for reconnects and call retries.
  */
 final class ExponentialBackoffPolicy implements BackoffPolicy {
-  static final class Provider implements BackoffPolicy.Provider {
+  /**
+   * Provider tuned for connection retries.
+   *
+   * https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
+   */
+  static final class ReconnectProvider implements BackoffPolicy.Provider {
+    @Override
+    public BackoffPolicy get() {
+      return new ExponentialBackoffPolicy(new Random(), TimeUnit.SECONDS.toMillis(1),
+              TimeUnit.MINUTES.toMillis(2),  1.6, .2);
+    }
+  }
+
+  /**
+   * Provider tuned for call retries.
+   */
+  static final class CallRetryProvider implements BackoffPolicy.Provider {
     @Override
     public BackoffPolicy get() {
-      return new ExponentialBackoffPolicy();
+      // TODO(lukaszx0) those values are picked arbirarily without much thought behind them. They
+      // need revising after testing call retries.
+      return new ExponentialBackoffPolicy(new Random(), 50, TimeUnit.SECONDS.toMillis(30), 1.2, .2);
     }
   }
 
-  private Random random = new Random();
-  private long initialBackoffMillis = TimeUnit.SECONDS.toMillis(1);
-  private long maxBackoffMillis = TimeUnit.MINUTES.toMillis(2);
-  private double multiplier = 1.6;
-  private double jitter = .2;
+  private final Random random;
+  private final long initialBackoffMillis;
+  private final long maxBackoffMillis;
+  private final double multiplier;
+  private final double jitter;
 
-  private long nextBackoffMillis = initialBackoffMillis;
+  private long nextBackoffMillis;
+
+  public ExponentialBackoffPolicy(Random random, long initialBackoffMilis, long maxBackoffMillis,
+                                  double multiplier, double jitter) {
+    this.random = random;
+    this.initialBackoffMillis = initialBackoffMilis;
+    this.maxBackoffMillis = maxBackoffMillis;
+    this.multiplier = multiplier;
+    this.jitter = jitter;
+
+    this.nextBackoffMillis =  initialBackoffMillis;
+  }
 
   @Override
   public long nextBackoffMillis() {
```

Comment:
It's unusual for 'policy' objects to be stateful like this. I would have expected the API to be

policy.backoff(lastBackoffMillis)